### PR TITLE
Bacteria Part 2: Evolution and Walls

### PR DESCRIPTION
**Pull request by Untrustedlife**
im submittng the pull request just so that we have one. Do not merge until i say its ready.
This is for issue #293

### CODE REVIEW

**Comment by Untrustedlife** [comment]
Note cell walls are mostly implemented, i just need to tweak some numbers here and  there at this point, and allow the angelscript to change membrane types (it may be useful to move membrane type to the species as a variable, and i may do that) Also, feel free to discuss bacteria and prokaryotic organelles and such here.

**Comment by hhyyrylainen** [comment]
I changed the target to be fix_everything as this should be merged there before that is merged to master.

File: scripts/microbe_stage/procedural_microbes.as [hunk 1]
```diff
@@ -6,11 +6,13 @@
 const auto MIN_INITIAL_LENGTH = 5;
 const auto MAX_INITIAL_LENGTH = 15;
 float MAX_CHANCE_SCORE = 0;
+float MAX_PROKARYOTE_SCORE = 0;
```
Comment (hhyyrylainen) [inline scripts/microbe_stage/procedural_microbes.as]: Why these aren't const (I hope these aren't used as global variables)?

File: scripts/microbe_stage/species_system.as [hunk 2]
```diff
@@ -327,8 +359,14 @@ const auto MIN_BACTERIA = 2;
 //! Updates the species's population and creates new ones. And keeps track of Species objects
 class SpeciesSystem : ScriptSystem{
 
+	int currentBacteriaAmount;
+	int currentEukaryoteAmount;
+
     void Init(GameWorld@ w){
 
+		currentBacteriaAmount=0;
```
Comment (hhyyrylainen) [inline scripts/microbe_stage/species_system.as]: To match existing style this initialization should be in the class (on line 362) and the variables should be moved to be after the methods where the other member variables are.

File: src/ThriveGame.cpp [hunk 3]
```diff
@@ -1201,6 +1201,27 @@ bool
...
+
+	if (engine->RegisterEnum("MEMBRANE_TYPE") < 0) {
+		ANGELSCRIPT_REGISTERFAIL;
+	}
+
+	ANGELSCRIPT_REGISTER_ENUM_VALUE(MEMBRANE_TYPE, membrane);
+	ANGELSCRIPT_REGISTER_ENUM_VALUE(MEMBRANE_TYPE, wall);
+	ANGELSCRIPT_REGISTER_ENUM_VALUE(MEMBRANE_TYPE, chitin);
+
+	if (engine->RegisterObjectMethod("MembraneComponent",
```
Comment (hhyyrylainen) [inline src/ThriveGame.cpp]: Return type should be changed to MEMBRANE_TYPE

File: src/microbe_stage/membrane_system.h [hunk 4]
```diff
@@ -14,6 +14,9 @@
 
 namespace thrive {
 
+//enumerable for membrane type
+enum class MEMBRANE_TYPE { membrane, wall, chitin };
```
Comment (hhyyrylainen) [inline src/microbe_stage/membrane_system.h]: These enum values should be UPPERCASE to match the style. (I'm not sure if this is in the style guide).

File: src/microbe_stage/membrane_system.h [hunk 5]
```diff
@@ -28,9 +31,18 @@ class MembraneComponent : public Leviathan::Component {
...
-    MembraneComponent();
+    MembraneComponent(MEMBRANE_TYPE type);
 	virtual ~MembraneComponent();
 
+
+
+	//holder for membrane type
+	MEMBRANE_TYPE membraneType;
+
+	void setMembraneType(MEMBRANE_TYPE type);
```
Comment (hhyyrylainen) [inline src/microbe_stage/membrane_system.h]: There should be a comment about this not taking affect without reseting this membrane as only that causes the mesh to be re-generated.

File: src/microbe_stage/membrane_system.h [hunk 6]
```diff
@@ -28,9 +31,18 @@ class MembraneComponent : public Leviathan::Component {
...
+
+	int getMembraneType();
```
Comment (hhyyrylainen) [inline src/microbe_stage/membrane_system.h]: Return type should be MEMBRANE_TYPE

**Comment by Untrustedlife** [comment]
Those changes are reasonable, i'll do it all when i get home tonight so we can merge it

File: scripts/microbe_stage/species_system.as (hunk 2 above)
Comment (Untrustedlife) [reply scripts/microbe_stage/species_system.as]: but this variable is seperate from any individual species

Comment (Untrustedlife) [reply scripts/microbe_stage/species_system.as]: actually i get what you mean nevermind

**Comment by Untrustedlife** [comment]
Also, those conflicts are simply there because you formatted the code on your branch, then the code was formatted on my branch, even though there are many many code changes which makes them look totally different to github because it looks like many things are being overwritten. (specifically whitespace) SO you can probabbly just merge then choose my branch for everything (I dont think you edited any of this code so it shouldn't cause issues, i would of course double check though :))

**Pull request merged**